bpf/go_common.h

```diff
@@ -105,6 +108,30 @@ struct {
     __uint(max_entries, MAX_CONCURRENT_REQUESTS);
 } ongoing_sql_queries SEC(".maps");
 
+typedef struct grpc_header_field {
+    u8 *key_ptr;
+    u64 key_len;
+    u8 *val_ptr;
+    u64 val_len;
+    u64 sensitive;
+} grpc_header_field_t;
+
+// assumes s2 is all lowercase
+static __always_inline int bpf_memicmp(const char *s1, const char *s2, s32 size) {
+    for (int i = 0; i < size; i++) {
+        if (s1[i] != s2[i] && s1[i] != (s2[i] - 32)) // compare with each uppercase character
+        {
+            return i + 1;
+        }
+    }
+
+    return 0;
+}
```

Comment:
```suggestion
static __always_inline int bpf_striequals(const char *s1, const char *s2, u32 size) {
    for (u32 i = 0; i < size; i++) {
        if (s1[i] != s2[i] && s1[i] != (s2[i] - 32)) // compare with each uppercase character
        {
            return 0;
        }
    }

    return 1;
}
```

rationale:
1. memcmp/strcmp return "less, equal, greater" whilst this function basically returns an inverted boolean on its original implementation - so I suggest renaming it to something completely unrelated to avoid confusion on the semantics
2. there's no reason for signed size

Reply:
I agree, but the idea was to have the same semantics as memcmp, as bad as they are.